scim/resource_service_principal.go

```diff
@@ -122,6 +122,10 @@ func ResourceServicePrincipal() *schema.Resource {
 				Type:     schema.TypeBool,
 				Optional: true,
 			}
+			m["disable_as_user_deletion"] = &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
```

Comment:
How come there is no `computed: ` value here like with users? How does the computed val get used?

Reply:
Computed did not help as I needed to be able to tell the difference if a user actually set a value or they are taking the default (i.e. null). Only if taking the default, do I compute the value.